### ISSUE TITLE
FileSequence.basename cut at the wrong place with still frames

### ISSUE THREAD

**Issue by mathbou** [sqlboy/fileseq#65]
I use `findSequencesOnDisk` to retrieve my sequences, it also returns the files without padding as I want it. But with the still frames, the `basename` methods cut the name at the first point, not the last.

I have this kind of filename: _PROJECT_010_110_CAT.RGB_color.exr_
The `basename` method returns: _PROJECT_010_110_CAT._
The `extension` method returns: _.RGB_color.exr_

With a filename with padding: _PROJECT_010_110_CAT.RGB_color.0042.exr_
The basename returned is: _PROJECT_010_110_CAT.RGB_color._
And the extension is: _.exr_

**Comment by justinfx** [comment]
I'm guessing this problem was introduced via the fix for #56. It was meant to support extensions with multiple dot components. But it would seem it is not properly ignoring patterns where there is no frame range. It shouldn't parse the longer extension in this case. I will try and look into this shortly.

**Comment by yawpitch** [comment]
Hmm, at first blush I like the ".tar.gz", but something tells me it's probably better to try and remain consistent with os.path.splitext and pathlib.Path.suffix.

**Comment by justinfx** [comment]
Yea. I thought it would be good to try and preserve things like that,
considering the use case. The problem is that if I don't capture it, where
do I store the extra string component without dropping it?


file.0100.tar.gz

If the extension is captured as .gz, then .tar is lost. It isn't as simple
as splitext since we store dirname, basename, range, pad, ext

**Comment by yawpitch** [comment]
Yeah, not sure. This is where pathlib.Path offers the .suffixes method, which gives you a list of all the extension components... I've wondered about rewriting FileSeq in light of pathlib, subclassing Path to incorporate the idea of frame and padding, but leaving the non-sequential extension components in .suffixes, but I was holding off until having time to rewrite for 3 as well.

**Comment by justinfx** [comment]
Well I was also thinking of rewriting fileseq to be a binding of libfileseq
(c++) so that I don't have to maintain the 3 individual language ports that
I currently look after. Also the gofileseq/libfileseq versions have a more
memory efficient solution to storing range data than the python
implementation.

**Comment by yawpitch** [comment]
Understandable, though that doesn't directly impact whether or not you'd want to have the Python version using pathlib.Path semantics to provide a more useful (and Pythonic) interface to that binding.

**Comment by justinfx** [comment]
I'm going to close this, as we have a `FilePathSequence` class option to specialize any user-defined behaviour around `pathlib`
